### DISCUSSION TITLE
[Feature] OpacityMaskView

### DISCUSSION THREAD

**Discussion by h82258652** [CommunityToolkit/Labs-Windows#490] [Experiments]
Provide a new control that can work like https://learn.microsoft.com/en-us/dotnet/api/system.windows.uielement.opacitymask?view=windowsdesktop-7.0

**Comment by niels9001** [comment]
@h82258652 Very interesting! Would you mind describing a bit what this control would do and how it would be used (in XAML)?

I know we did something similiar a while back in the WinUI Gallery: https://github.com/microsoft/WinUI-Gallery/blob/main/WinUIGallery/Controls/HomePageHeaderImage.xaml.cs

Having some sort of brush/control would be amazing and so much easier to use!

**Reply by h82258652** [reply]
Hi @niels9001 , I created a PR #491 for this.

It will be used like this in XAML:
```XAML
<controls:OpacityMaskView>
    <controls:OpacityMaskView.OpacityMask>
        <LinearGradientBrush StartPoint="0,0" EndPoint="1,0">
            <GradientStop Offset="0" Color="White" />
            <GradientStop Offset="1" Color="Transparent" />
        </LinearGradientBrush>
        <!--  or other brush  -->
    </controls:OpacityMaskView.OpacityMask>
    <Button Content="Hello world" />
</controls:OpacityMaskView>
```

**Reply by michael-hawker** [reply]
I see, so this is basically a polyfill for this WPF feature? Interesting. 🤔
